Questions.ViewModelComposition/QuestionsModule.cs

```diff
@@ -0,0 +1,75 @@
+﻿using ITOps.Json;
+using ITOps.ViewModelComposition.Modules;
+using Questions.ViewModelComposition.Events;
+using System.Collections.Generic;
+using System.Dynamic;
+using System.Linq;
+using System.Net.Http;
+
+namespace Questions.ViewModelComposition
+{
+    public class QuestionsModule : RequestsModule
+    {
+        public QuestionsModule()
+        {
+            Get("/questions", async (requestId, vm) =>
```

Comment:
The syntax looks nice, but I don't like that this is defined in the constructor. Potential solutions:
- Move the lambda to a method, so in the constructor you only have Get("/questions", GetList)
- Move each lambda to a separate IHandleRequest class and do Get<GetListHandler>("/questions")
- The previous point would lead to have the IHandleRequest auto discovered with attribute routing defined there

Reply:
I went for the same style used by Nancy https://github.com/NancyFx/Nancy/blob/master/samples/Nancy.Demo.Authentication.Basic/MainModule.cs#L5

The lambda can be defined where ever you like, that doesn't really matter.

Discovering requires reflection, that will have an impact on startup time, that's why Nancy went that route. Attribute routing can be supported as well

Reply:
OK. If you don't do autodiscovery, attribute routing doesn't add much IMO. With this Module approach, I'd personally prefer isolating the definitions from the handlers, like just doing `Get<THandler>("route");` in the Module and define the THandlers separately similar to NSB. They could also receive some context as a parameter instead of using Module base class methods like in Nancy's Bind, Request, etc. It'd improve testability a lot IMO.

Reply:
My plan/idea is to let users define what they want, e.g.:

```
services.AddViewModelComposition()
   .EnableAttributeRouting() //MVC Style routing
   .EnableModules();  //Nancy style
```

In this way you could use the flavor you prefer, and even mix them as you like in order to achieve different things.